packages/manager/src/components/authentication/APIKeyCreateModal.tsx

```diff
@@ -0,0 +1,111 @@
+import React, { useState, useEffect } from "react";
+import { ActionGroup, Button, Checkbox, Modal, Form, FormGroup, TextInput } from "@patternfly/react-core";
+import * as uuid from "uuid";
+
+interface IProps {
+  isOpen: boolean;
+  onClose: () => void;
+}
+export default (props: IProps) => {
+  const { isOpen, onClose } = props;
+  const [name, setName] = useState("");
+  const [expired, setExpired] = useState("");
+  const [scopes, setScopes] = useState<string[]>([]);
+  const [apiKey, setApiKey] = useState("");
+
+  useEffect(() => {
+    setName("");
+    setExpired("");
+    setApiKey("");
+    setScopes([]);
+  }, [isOpen]);
+
+  const onSubmit = () => {
+    setApiKey(uuid.v4());
+  };
+
+  const isValid = () => {
+    if (name && name.trim().length > 0 && scopes.length > 0) {
+      return true;
+    }
+    return false;
+  };
+
+  const onChangeScope = (checked: boolean, event: React.FormEvent<HTMLInputElement>) => {
+    const env = event.currentTarget.name;
+    console.log(env);
+    if (!checked) {
+      setScopes(scopes.filter(scope => scope !== env));
+    } else {
+      setScopes([...scopes, env]);
+    }
+  };
+
+  const envs = ["Dev", "QA", "Stage", "Prod"];
```

Comment:
You don't have to hardcode the envs, they're available from EnvParser.  They should reflect the environments that the user has passed in to `REACT_APP_SPASHIP_URLS`.

Reply:
Yes, I have plan in next PR to fix all UI which use environment

Reply:
I'll make the change in next commit of this PR